runtime/shared/shrclssup.c

```diff
@@ -303,6 +303,10 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void* reserved)
 					return rc;
 				}
 			}
+			/* New GC API j9gc_updateHeapSizes() calls vm->sharedClassConfig->findGCHints() to find the previously stored GC hints
+			 * from the share cache, and then decide whether to use the hints. This call can also be put inside GC.
+			 */
+			/* vm->memoryManagerFunctions->j9gc_updateHeapSizes(); */
```

Comment:
I believe it's better that GC just calls findGCHints() at appropriate moment during GC initialization sequence (late enough when we know that sharedcache is up and readable, but early enough before we started creating heap).

If so, findGCHints() must be a cross module call (which seems like it already is).

Reply:
> I believe it's better that GC just calls findGCHints() at appropriate moment during GC initialization sequence

I agree.